### PR TITLE
As a developer, you want to provide your signed-in users an option to get and present Verifiable credentials.

### PR DESCRIPTION
**Pull request by premgan**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by premgan — COMMENTED** [review]
Should we consider adding a Deployment section like here ?
https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/master/2-WebApp-graph-user/2-1-Call-MSGraph/README.md#deployment

File: 2-asp-net-core-api-user-signin/README.md [hunk 1]
```diff
@@ -94,33 +111,43 @@ If you want to modify the payloads `issuance_request_config.json` and `presentat
...
+Make sure you copy the `ClientId` and `TenantTd` you copied when creating the app registration to the `appsettings.json` **AzureAd** section as well.
+
 ## Running the sample
 
-1. Open a command prompt and run the following command:
+1. Open a command prompt, run ngrok to set up a URL on 3000. You can install ngrok globally by using the [ngrok npm package](https://www.npmjs.com/package/ngrok/).
 ```Powershell
-dotnet build "AspNetCoreVerifiableCredentials.csproj" -c Debug -o .\bin\Debug\net5
-dotnet run
+ngrok http 3000
```
Comment (premgan) [inline 2-asp-net-core-api-user-signin/README.md:120]: Alternatively if we want to improve the directions even for future samples where multiple registrations are needed, we can recommend using paid permanent ngrok URL. Something like below.. 

==
Open a command prompt, run ngrok to expose your local web server running on port 3000 via a public URL. For example: 
    * Win: `./ngrok http 3000 -host-header=localhost:3000 -subdomain="contoso"`
    * Mac: `/ngrok http 3000 -host-header=localhost:3000 -subdomain="contoso"
    
`NOTE: The free ngrok plan will generate a new URL every time you run it, which requires you to follow Step 2 below every time you run the sample. A paid account with a permanent ngrok URL is recommended.
==

File: 2-asp-net-core-api-user-signin/IssuerController.cs [hunk 2]
```diff
@@ -116,14 +116,24 @@ public async Task<ActionResult> IssuanceRequest()
...
                     string host = GetRequestHostName();
                     if (!host.Contains("//localhost"))
                     {
                         payload["callback"]["url"] = String.Format("{0}:/api/issuer/issuanceCallback", host);
                     }
+                    else
+                    {
+                        if (!String.IsNullOrWhiteSpace(AppSettings.VC_CallbackHostURL))
+                        {
+                            payload["callback"]["url"] = String.Format("{0}:/api/issuer/issuanceCallback", AppSettings.VC_CallbackHostURL);
```
Comment (mahoekst) [inline 2-asp-net-core-api-user-signin/IssuerController.cs:130]: Can we remove the : behind {0}

File: 2-asp-net-core-api-user-signin/Properties/launchSettings.json [hunk 3]
```diff
@@ -18,10 +18,10 @@
     "AspNetCoreVerifiableCredentials": {
       "commandName": "Project",
       "launchBrowser": true,
-      "applicationUrl": "http://0.0.0.0:5000",
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
-      }
+      },
+      "applicationUrl": "http://0.0.0.0:3000"
```
Comment (mahoekst) [inline 2-asp-net-core-api-user-signin/Properties/launchSettings.json:24]: The other sample runs on port 5000, can you revert back to 5000

File: 2-asp-net-core-api-user-signin/appsettings.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
     "CertificateName": "[Or instead of client secret: Enter here the name of a certificate (from the user cert store) as registered with your application]",
     "IssuerAuthority": "YOUR VC SERVICE DID",
     "VerifierAuthority": "YOUR VC SERVICE DID",
-    "CredentialManifest": "THE CREDENTIAL URL FROM THE VC PORTAL"
+    "CredentialManifest": "THE CREDENTIAL URL FROM THE VC PORTAL",
+    "VC_CallbackHostURL": "Please refer to README section on Running the sample for instructions on how to set this value."
```
Comment (mahoekst) [inline 2-asp-net-core-api-user-signin/appsettings.json:30]: remove the _ to allign with the rest of the setting names

File: README.md [hunk 5]
```diff
@@ -6,6 +6,7 @@ This repo contains a set of Azure AD Verifiable Credentials samples
 | Sample | Description |
 |------|--------|
 | 1-asp-net-core-api-idtokenhint | dotnet sample for using the VC Request API to issue and verify verifiable credentials with a credential contract which allows the VC Request API to pass in a payload for the Verifiable Credentials|
+| 2-asp-net-core-api-user-signin | dotnet sample for a Line of Business (LOB) developer who wants to provide the signed-in users an option to get and present Verifiable credentials using the VC Request API. **Note:** This is different from 1-asp-net-core-api-idtokenhint sample as follows : User sign-in is a requirement to issue credentials since the credentials have claims (first name, last name) based on the signed-in user's idToken.'|
```
Comment (mahoekst) [inline README.md:9]: Can you remove the lines which talk about a Line of Business developer. We didn't define audiences for the sample. This is just a sample where a user can sign-in and information from the ID token is used as input for the payload and issue a VC

**Pull request merged**